Index: source/dorm.INC
===================================================================
--- source/dorm.INC	(revision 167)
+++ source/dorm.INC	(working copy)
@@ -9,8 +9,8 @@
   define a comment.
 }
 {$DEFINE LINK_FIREBIRDUIB_ADAPTER }
-{$DEFINE LINK_INTERBASEUIB_ADAPTER }
-{$DEFINE LINK_SQLITE3_ADAPTER}
+{.$DEFINE LINK_INTERBASEUIB_ADAPTER }
+{.$DEFINE LINK_SQLITE3_ADAPTER}
 
 // requires delphi ent or better - NOT YET SUPPORTED
 { . $DEFINE LINK_SQLSERVER_ADAPTER }
Index: source/dorm.pas
===================================================================
--- source/dorm.pas	(revision 167)
+++ source/dorm.pas	(working copy)
@@ -1852,8 +1852,11 @@
     ParentObject := TdormUtils.GetProperty(AObject, _belongs_to.Name).AsObject;
     if assigned(ParentObject) then
     begin
-      _parent_type := FCTX.FindType(Qualified(AMappingTable,
-        _belongs_to.OwnerClassName));
+      _parent_type := _belongs_to.RTTICache.RTTIProp.PropertyType;
+      //_parent_type := FCTX.FindType(_belongs_to.RTTICache.RTTIProp.PropertyType.QualifiedName);
+
+//      _parent_type := FCTX.FindType(Qualified(AMappingTable,
+//        _belongs_to.OwnerClassName));
       if not assigned(_parent_type) then
         raise Exception.Create('Unknown type ' + _belongs_to.OwnerClassName);
       v := GetIdValue(FMappingStrategy.GetMapping(_parent_type).Id,
